File: typescript/resistor-color-duo/resistor-color-duo.ts
export const colorValues = new Map<string, number>([
  ["black", 0],
  ["brown", 1],
  ["red", 2],
  ["orange", 3],
  ["yellow", 4],
  ["green", 5],
  ["blue", 6],
  ["violet", 7],
  ["grey", 8],
  ["white", 9],
]);

export function decodedValue(colors: string[]): number {
  let result = 0;
  let position = 10;
  for (const color of colors.slice(0, 2)) {
    result += position * colorValues.get(color)!;
    position /= 10;
  }
  return result;
}

File: zig/resistor-color-duo/resistor_color_duo.zig
const std = @import("std");
pub const ColorBand = enum {
    black,
    brown,
    red,
    orange,
    yellow,
    green,
    blue,
    violet,
    grey,
    white,
};

pub fn colorCode(colors: []const ColorBand) anyerror!isize {
    comptime {
        // tens position
        const a: isize = @intCast(isize, 10) * @enumToInt(colors[0]);
        // ones position
        const b: isize = @enumToInt(colors[1]);
        return a + b;
    }
}
